src/pymodaq_plugins_keithley/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Keithley27XX.py

```diff
@@ -0,0 +1,248 @@
+    :type params: dictionnary list
+
+    :param x_axis: Daq_0Dviewer doesn't have axis, data acquired sample by sample
+    :type x_axis: Nonetype
+    """
+    all_config = {}
+    current_config = None
+    resources_path = plugin.__path__[0] + "/resources"
+
+    # Configurations for supported Keithley instruments
+    toml_keithley = [f for f in os.listdir(resources_path) if "keithley.toml" in f]
+    all_config["base"] = plugin.config_keithley
+
+    # Configurations for supported Keithley switching modules
+    toml_modules = [f for f in os.listdir(resources_path) if "module" in f and ".toml" in f]
+    for file in toml_modules:
+        exec("all_config[" + str(file[-9:-5]) + "] = plugin.config_k" + str(file[-9:-5]))
+        if current_config == None:
+            current_config = all_config.get(int(file[-9:-5]))
+
+    # Read configuration file
+    panel = all_config["base"]('INSTRUMENT').get('panel').upper()
+    resources_list = [rsrc("INSTRUMENT").get("rsrc_name") for key, rsrc in list(all_config.items()) if key != "base"]
+    print("resources list = ", resources_list)
+
+    if panel == 'FRONT':
+        params = comon_parameters + [
+            {'title': 'Resources', 'name': 'resources', 'type': 'list', 'limits': resources_list,
+             'value': resources_list[0]},
+            {'title': 'Keithley', 'name': 'Keithley_Params', 'type': 'group', 'children': [
+                {'title': 'ID', 'name': 'ID', 'type': 'text', 'value': ''},
+                {'title': 'FRONT panel', 'name': 'frontpanel', 'type': 'group', 'children': [
+                    {'title': 'Mode', 'name': 'frontmode', 'type': 'list',
+                     'limits': ['VOLT:DC', 'VOLT:AC', 'CURR:DC', 'CURR:AC', 'RES', 'FRES', 'FREQ', 'TEMP'],
+                     'value': 'VOLT:DC'}]}
+            ]}
+        ]
+    elif panel == 'REAR':
+        params = comon_parameters + [
+            {'title': 'Resources', 'name': 'resources', 'type': 'list', 'limits': resources_list,
+             'value': resources_list[0]},
+            {'title': 'Keithley', 'name': 'Keithley_Params', 'type': 'group', 'children': [
+                {'title': 'ID', 'name': 'ID', 'type': 'text', 'value': ''},
+                {'title': 'REAR panel', 'name': 'rearpanel', 'type': 'group', 'children': [
+                    {'title': 'Mode', 'name': 'rearmode', 'type': 'list',
+                     'limits': ['SCAN_LIST', 'VOLT:DC', 'VOLT:AC', 'CURR:DC', 'CURR:AC', 'RES', 'FRES', 'FREQ', 'TEMP'],
+                     'value': 'SCAN_LIST'}
+                ]}
+            ]}
+        ]
+
+    def __init__(self, parent=None, params_state=None):
+        super(DAQ_0DViewer_Keithley27XX, self).__init__(parent, params_state)
+
+    def ini_attributes(self):
+        """Attributes init when DAQ_0DViewer_Keithley class is instancied"""
+        print("Panel configuration 0D_viewer: ", self.panel)
+        self.channels_in_selected_mode = None
```

Comment:
add here
self.controller: Keithley = None

Reply:
Done. I didn't get the point of setting the instance variable in this way. It seems to me that the self.controller instance attribute remains of NoneType, so what is the difference with self.controller = None ?

Reply:
autocompletion !!!